### PR TITLE
Misuv 7841 Refactor income source end date controller

### PR DESCRIPTION
**Pull request by tioadesola1**
Everything refactored, all unit tests run. Acceptance tests has some fails but found to be unrelated to this ticket.

### CODE REVIEW

File: app/controllers/manageBusinesses/cease/DeclareIncomeSourceCeasedController.scala [hunk 1]
```diff
@@ -139,9 +139,9 @@ class DeclareIncomeSourceCeasedController @Inject()(val authorisedFunctions: Fro
     if (isAgent) routes.CeaseIncomeSourceController.showAgent()
     else         routes.CeaseIncomeSourceController.show()
 
-  private val redirectAction: (Option[String], Boolean, IncomeSourceType) => Call = (id, isAgent, incomeSourceType) =>
-    if (isAgent) routes.IncomeSourceEndDateController.showAgent(id, incomeSourceType)
-    else         routes.IncomeSourceEndDateController.show(id, incomeSourceType)
+  private val redirectAction: (Option[String], Boolean, Boolean, IncomeSourceType) => Call = (id, isAgent, isChange, incomeSourceType) =>
+    if (isAgent) routes.IncomeSourceEndDateController.show(id, incomeSourceType, isAgent, isChange)
```
Comment (avelx) [inline app/controllers/manageBusinesses/cease/DeclareIncomeSourceCeasedController.scala:143]: if control flow statement is not required here, 
you can just call in one line:
`routes.IncomeSourceEndDateController.show(id, incomeSourceType, isAgent, isChange)`

File: app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala [hunk 2]
```diff
@@ -62,10 +62,10 @@ class IncomeSourceEndDateController @Inject()(val authorisedFunctions: FrontendA
     val hashedId: Option[String] = maybeIncomeSourceId.map(_.toHash.hash)
 
     (isAgent, isChange) match {
-      case (false, false) => routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType)
-      case (false, _) => routes.IncomeSourceEndDateController.submitChange(hashedId, incomeSourceType)
-      case (_, false) => routes.IncomeSourceEndDateController.submitAgent(hashedId, incomeSourceType)
-      case (_, _) => routes.IncomeSourceEndDateController.submitChangeAgent(hashedId, incomeSourceType)
+      case (false, false) => routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType, isAgent, isChange)
```
Comment (avelx) [inline app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala:65]: the whole pattern matching here become redundant as you can simply call one line:
`routes.IncomeSourceEndDateController.submit(hashedId, incomeSourceType, isAgent, isChange)`

File: app/controllers/manageBusinesses/cease/DeclareIncomeSourceCeasedController.scala (hunk 1 above)
Comment (tioadesola1) [reply app/controllers/manageBusinesses/cease/DeclareIncomeSourceCeasedController.scala]: Made the corrections and removed statement from here

File: app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala (hunk 2 above)
Comment (tioadesola1) [reply app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala]: Removed the redundant pattern matching

**Comment by avelx** [comment]
@tioadesola1  - can you fix IT test failures pls ?

**Comment by tioadesola1** [comment]
> @tioadesola1 - can you fix IT test failures pls ?

All IT tests are fixed and passing now

**Review by avelx — APPROVED** [review]

**Comment by avelx** [comment]
> Everything refactored, all unit tests run. Acceptance tests has some fails but found to be unrelated to this ticket.

you git feature branch were behind main, so you have to merge main into your branch to to fix ATs failures;

**Pull request merged**